ariadne/contrib/federation/utils.py

```diff
@@ -79,15 +80,20 @@ def resolve_entities(_: Any, info: GraphQLResolveInfo, **kwargs) -> Any:
             type_object, "__resolve_reference__", lambda o, i, r: reference,
         )
 
-        result.append(
-            add_typename_to_possible_return(
-                resolve_reference(type_object, info, reference), __typename,
-            ),
-        )
+        representation = resolve_reference(type_object, info, reference)
+
+        if isawaitable(representation):
```

Comment:
How about:

```python
if isawaitable(representation):
    representation = await representation
result.append(add_typename_to_possible_return(representation, __typename))
```

Reply:
Nvm, this function cannot be asynchronous as it's also called in the synchronous execution flow.